iothub/service/src/Http2/HttpClientFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Net.Http;
+
+namespace Microsoft.Azure.Devices.Http2
+{
+    internal class HttpClientFactory
+    {
+        private const string HttpsEndpointPrefix = "https";
+
+        internal static HttpClient Create(string hostName, HttpTransportSettings2 settings)
+        {
+            if (settings.HttpClient != null)
+            {
+                return settings.HttpClient;
+            }
+
+            Uri httpsEndpoint = new UriBuilder(HttpsEndpointPrefix, hostName).Uri;
+#pragma warning disable CA2000 // Dispose objects before losing scope
+
+#if NETCOREAPP && !NETCOREAPP2_0 && !NETCOREAPP1_0 && !NETCOREAPP1_1
```

Comment:
I've removed netcoreapp as a target. I realized I didn't check for these directives. I'll go clean them up.